src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -317,18 +317,20 @@ internal void Connect(ServerInfo serverInfo, SqlInternalConnectionTds connHandle
                 Debug.Assert(false, "SNI returned status != success, but no error thrown?");
             }
 
-#if !MANAGED_SNI
+
             if (integratedSecurity)
             {
+#if !MANAGED_SNI
                 LoadSSPILibrary();
+#endif
                 // now allocate proper length of buffer
                 _sniSpnBuffer = new byte[SNINativeMethodWrapper.SniMaxComposedSpnLength];
```

Comment:
For Managed SNI, this buffer needn't be preinitialized. The CreateConnectionHandle would take care of initializing the _sniSpnBuffer.